### PR TITLE
8298075: RISC-V: Implement post-call NOPs

### PR DESCRIPTION
**Pull request by zhengxiaolinX**
Same as [JDK-8287567](https://bugs.openjdk.org/browse/JDK-8287567), we need to implement post-call NOPs in the RISC-V backend.

We use `li32` to implement it. `li32` will signed-extend the loaded imm32 to a 64-bit register. However, our reader `NativePostCallNop::displacement()` only needs the low 32 bits value so we are safe.

The current post-call NOP is like:
```
nop                    (always 4-byte, used for deoptimization, please see `NativePostCallNop::make_deopt()`)
lui zr, <imm20>        (always 4-byte)
addiw zr, zr, <imm12>  (always 4-byte)
```

The code size here is the same as the AArch64 counterpart, 12 bytes.

The logic of setting the post-call NOP displacement is at `install_post_call_nop_displacement()` [1] when nmethods are being installed, so no one is running post-call NOPs when we are patching them.

One can check them by directly using `java --enable-preview -XX:+UnlockDiagnosticVMOptions -XX:+PrintAssembly`.

Tested hotspot tier1\~4 and jdk tier1\~4 using fastdebug by force turning on `--enable-preview`; SPECjbb2015 and dacapo were tested without regressions found. Skynet JMH benchmark seems to have an improvement from `8662ms/op -> 8495ms/op`.

Thanks,
Xiaolin

[1] https://github.com/openjdk/jdk/blob/b49fd920b6690a8b828c85e45c10e5c4c54d2022/src/hotspot/share/code/nmethod.cpp#L1078-L1107

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8298075](https://bugs.openjdk.org/browse/JDK-8298075): RISC-V: Implement post-call NOPs

### Reviewers
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)
 * [Ludovic Henry](https://openjdk.org/census#luhenry) (@luhenry - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11502/head:pull/11502` \
`$ git checkout pull/11502`

Update a local copy of the PR: \
`$ git checkout pull/11502` \
`$ git pull https://git.openjdk.org/jdk pull/11502/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11502`

View PR using the GUI difftool: \
`$ git pr show -t 11502`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11502.diff">https://git.openjdk.org/jdk/pull/11502.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/riscv/nativeInst_riscv.cpp [hunk 1]
```diff
@@ -444,8 +444,16 @@ void NativePostCallNop::make_deopt() {
   NativeDeoptInstruction::insert(addr_at(0));
 }
 
+int NativePostCallNop::displacement() const {
+  // discard the high 32-bit
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/nativeInst_riscv.cpp:448]: Suggestion: // Discard the high 32 bits

Comment (zhengxiaolinX) [reply src/hotspot/cpu/riscv/nativeInst_riscv.cpp]: Thank you - done.

**Review by RealFYang — APPROVED** [review]
Change looks good. I also performed several benchmark tests (Dacapo, specjbb,2015, Renaissance) for this and I didn't notice regressions from the numbers. So I guess it should be safe to have this.

**Review by luhenry — APPROVED** [review]

**Comment by zhengxiaolinX** [comment]
Thank you for reviewing @RealFYang @luhenry and the additional testing!

/integrate

**Pull request closed without merging**